app/hooks/use-fetch-delegation-status.ts

```diff
@@ -1,19 +1,20 @@
-import { useCallback } from 'react';
-import useSWR from 'swr';
+import { useQuery } from 'react-query';
 import { cvToHex, standardPrincipalCV, tupleCV } from '@stacks/transactions';
 
 import { useApi } from './use-api';
 
 export function useFetchDelegationStatus(contractId?: string, address?: string) {
   const api = useApi();
 
-  const delegationFetcher = useCallback(
-    (_, contractId, address) => {
-      if (!contractId || !address) throw new Error('values not defined');
-      const [contractAddress, contractName] = contractId.split('.');
+  const { data, refetch } = useQuery(
+    ['delegation-status', contractId, address],
```

Comment:
small nit, i prefer to have query keys as consts, eg `const QUERY_DELEGATION_STATUS = 'delegation-status';` but def not blocking, just personal preference.

Reply:
Agree, I was thinking of even using an enum

Reply:
Yep I think an enum would be ideal

Reply:
done